### PR TITLE
fix gpus -1 for CPU

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

the `gpus=-1` shall be robust even if you do not have any GPU

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ethanwharris — COMMENTED** [review]
Maybe update a CPU test to use `gpus=-1` to show that it is working

File: pytorch_lightning/utilities/device_parser.py [hunk 1]
```diff
@@ -184,11 +184,13 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
...
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus() -> Optional[List[int]]:
     """
     Returns:
          a list of all available gpus
     """
+    if not torch.cuda.is_available():
+        return None
```
Comment (awaelchli) [inline pytorch_lightning/utilities/device_parser.py:193]: Hey @Borda, explicitly asking for gpus and silently not getting them when on a CPU-only machine may be problematic. What do you say?

Comment (Borda) [reply pytorch_lightning/utilities/device_parser.py:193]: it is not silently, you just say give me the best, so if you have GPU, give me all, if you run the same on a notebook, do not crash, as it is not with `-1` I can't run the same code anywhere a need to change it...

Comment (Borda) [reply pytorch_lightning/utilities/device_parser.py:193]: @awaelchli this is the usecase in Flash, I wanted to use `-1` but it would crash so I have to use `torch.cuda.device_count()`, https://github.com/PyTorchLightning/lightning-flash/pull/638

Comment (awaelchli) [reply pytorch_lightning/utilities/device_parser.py:193]: Actually I think it's ok since we print GPU available: True/False GPU used: True/False with every lIghtning training. 
so adding a warning would not do much more. 

don't forget to update tests :)

**Review by awaelchli — APPROVED** [review]
would be nice to include this change in the changelog :)

**Comment by awaelchli** [comment]
I would also like to suggest not classifying this as bugfix :) 
We have documented the current behavior here: https://pytorch-lightning.readthedocs.io/en/latest/advanced/multi_gpu.html#select-gpu-devices

File: pytorch_lightning/utilities/device_parser.py (hunk 1 above)
Comment (carmocca) [inline pytorch_lightning/utilities/device_parser.py:193]: should this return instead

```suggestion
        return []
```

Comment (Borda) [reply pytorch_lightning/utilities/device_parser.py:193]: that is what it does now, so maybe we shall fix it elsewhere?
```py
list(range(0)) # -> []
```

Comment (carmocca) [reply pytorch_lightning/utilities/device_parser.py:193]: Yeah, that would work too. Worth investigating

File: pytorch_lightning/utilities/device_parser.py [hunk 2]
```diff
@@ -184,11 +184,13 @@ def _normalize_parse_gpu_input_to_list(gpus: Union[int, List[int], Tuple[int, ..
     return list(range(gpus))
 
 
-def _get_all_available_gpus() -> List[int]:
+def _get_all_available_gpus() -> Optional[List[int]]:
```
Comment (tchaton) [inline pytorch_lightning/utilities/device_parser.py:187]: Let's add a test to prevent regression.
